diff --git a/Source/FactoryGame/Public/FGOptionsSettings.h b/Source/FactoryGame/Public/FGOptionsSettings.h
index 4fc16b38a..f878e244c 100644
--- a/Source/FactoryGame/Public/FGOptionsSettings.h
+++ b/Source/FactoryGame/Public/FGOptionsSettings.h
@@ -3,6 +3,8 @@
 #pragma once
 
 #include "OptionValueContainer.h"
+#include "Engine/DeveloperSettings.h"
+
 #include "FGOptionsSettings.generated.h"
 
 UENUM( BlueprintType )
diff --git a/Source/FactoryGame/Public/FGSaveSession.h b/Source/FactoryGame/Public/FGSaveSession.h
index d0d87de9a..051da2e2f 100644
--- a/Source/FactoryGame/Public/FGSaveSession.h
+++ b/Source/FactoryGame/Public/FGSaveSession.h
@@ -5,6 +5,7 @@
 #include "FactoryGame.h" // MODDING EDIT: no PCH
 #include "UObject/Object.h"
 #include "FGSaveSystem.h"
+#include "Serialization/BufferArchive.h" //MODDING EDIT: SaveToDiskWithCompression wants FBufferArchive
 #include "FGObjectReference.h"
 #include "FGSaveSession.generated.h"
 
diff --git a/Source/FactoryGame/Public/Hologram/FGBuildableHologram.h b/Source/FactoryGame/Public/Hologram/FGBuildableHologram.h
index 5a088bdfe..c7f7be54b 100644
--- a/Source/FactoryGame/Public/Hologram/FGBuildableHologram.h
+++ b/Source/FactoryGame/Public/Hologram/FGBuildableHologram.h
@@ -16,6 +16,10 @@ UCLASS()
 class FACTORYGAME_API AFGBuildableHologram : public AFGHologram
 {
 	GENERATED_BODY()
+
+	// User Mod Edits
+	friend class FFicsItNetworksModule;
+	// User Mod Edits
 public:
 	/** Tags for marking components in the hologram. */
 	static FName mInputConnectionMeshTag;
diff --git a/Source/FactoryGame/Public/ItemAmount.h b/Source/FactoryGame/Public/ItemAmount.h
index 19438189f..aa627d5f3 100644
--- a/Source/FactoryGame/Public/ItemAmount.h
+++ b/Source/FactoryGame/Public/ItemAmount.h
@@ -4,6 +4,7 @@
 
 #include "FactoryGame.h" // MODDING EDIT: no PCH
 #include "Resources/FGItemDescriptor.h"
+#include "FactoryGame.h" // MODDING EDIT
 #include "ItemAmount.generated.h"
 
 /**
diff --git a/Source/FactoryGame/Public/Resources/FGItemDescriptor.h b/Source/FactoryGame/Public/Resources/FGItemDescriptor.h
index 3e9e5a5b8..b498b779d 100644
--- a/Source/FactoryGame/Public/Resources/FGItemDescriptor.h
+++ b/Source/FactoryGame/Public/Resources/FGItemDescriptor.h
@@ -3,6 +3,7 @@
 #pragma once
 
 #include "UObject/Object.h"
+#include "FactoryGame.h" // MODDING EDIT
 #include "Styling/SlateBrush.h"
 #include "FGItemDescriptor.generated.h"
 
diff --git a/Source/FactoryGame/Public/SharedInventoryStatePtr.h b/Source/FactoryGame/Public/SharedInventoryStatePtr.h
index bfdfe9eeb..c05cfb358 100644
--- a/Source/FactoryGame/Public/SharedInventoryStatePtr.h
+++ b/Source/FactoryGame/Public/SharedInventoryStatePtr.h
@@ -3,6 +3,7 @@
 #pragma once
 
 #include "FGSaveSession.h"
+#include "FactoryGame.h" // MODDING EDIT
 #include "GameFramework/Actor.h"
 #include "Templates/SharedPointer.h"
 #include "Engine/PackageMapClient.h"
